provision/provision.sh

```diff
@@ -18,6 +18,14 @@ start_seconds="$(date +%s)"
 # fix no tty warnings in provisioner logs
 sudo sed -i '/tty/!s/mesg n/tty -s \\&\\& mesg n/' /root/.profile
 
+# fix commands from homebin not working with sudo, clean first and then append at the end
```

Comment:
What is this command doing? 🤔

Reply:
Quite cryptic indeed, maybe is taking all the aliases and putting that inside the sudoers file?

Reply:
The sed is editing a setting for the sudo command, which basically resets the PATH when you do `sudo command`. If you do `sudo su` then you're logged in as root, and the bash_aliases is loaded normally from the root profile. 

But when you run a command with `sudo` (which seems to be the way that scripts are executed by vagrant) then the path used is the one on the `secure_path` setting of the `/etc/sudoers` file. More info on `man sudoers` or https://www.sudo.ws/man/1.8.15/sudoers.man.html

What this command does is check for the presence of the hombin path (at the beggining or anywhere else), remove it, and then append it at the end of that same setting. 

If the setting (for whatever reason) doesn't exist, then the `sed` would do nothing. Also, this sed syntax basically executes 3 expressions on one run (to avoid multiple seds).

The reason I'm then sourcing the bash_aliases into the current shell, is because the change on the sudoers file doesn't apply until you restart the current shell.

Does this make a bit more sense now?

Reply:
Thanks, that's a lot clearer :)